### PR TITLE
Phase 2: harden and cover (re-enable tests, align coverage exclusions, clean warnings)

### LINKED ISSUE #2975: Finish unit-test coverage for GetConsentStatusChanges (consent receiver listing)
## Problem

Follow-up for [#2959](https://github.com/Altinn/altinn-authorization-tmp/pull/2959): the unit tests added there cover roughly half of the cases identified for the new code under `feature/get-consent-changes-list`. The remaining cases are scaffolded as TODO comments inline with the existing tests so the next iteration is mostly fill-in-the-blank.

The `GetConsentStatusChanges` receiver-listing path has a coverage gap across the extension, service, controller, and repository layers.

## Goal

Fill in the remaining identified test cases across all four layers so the `GetConsentStatusChanges` receiver-listing code is covered.

### Extension method — `ConsentStatusChangeExtensions.ToDto`

File: `src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Helpers/Extensions/ConsentStatusChangeExtensionsTests.cs`

- [ ] `ToDto_ForDeletedExpiredOrUsedEvent_SerialisesEnumNameAsString` — extend the existing `[Theory]` with `Deleted`, `Expired`, `Used`. Keep "currently surfaced" cases separate from "defensive" ones so a wire-format change pinpoints the right group.
- [ ] `ToDto_PreservesChangedDateTimeZoneOffset` — non-UTC `ChangedDate` (e.g. `+02:00`) must round-trip without normalising to UTC.
- [ ] `ToDto_DoesNotSurfaceConsentEventId` — pin the contract that `ConsentEventId` stays internal (used only to build the pagination cursor). Reflection-based assertion is fine.

### Service — `ConsentService.GetConsentStatusChangesForParty`

File: `src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Services/ConsentServiceTests.cs`

- [ ] `GetConsentStatusChangesForParty_PersonIdReceiver_ResolvesViaGetByPersonNo` — mirror of the existing `OrganizationId` test but for `PersonIdentifier`. Confirms the other branch of `MapFromExternalIdenity`.
- [ ] `GetConsentStatusChangesForParty_RepositoryReturnsEmptyList_ReturnsOkWithEmptyList` — controller materialises this into a paginated response with no `next` link, so an empty (not null) collection matters.
- [ ] `GetConsentStatusChangesForParty_PassesContinuationTokenAndPageSizeVerbatim` — `[Theory]` with representative tokens (`null`, `""`, base64 cursor) and page sizes (`1`, `100`, `1000`). Page-size clamping lives in the *repository*; the service must not mutate these.
- [ ] `GetConsentStatusChangesForParty_CancellationTokenForwarded` — pass a real `CancellationTokenSource.Token` and verify the repo received the same token via `It.Is<CancellationToken>(t => t == ct)`. Catches "default-token" regressions.
- [ ] **(needs team discussion)** `GetConsentStatusChangesForParty_AmPartyServiceReturnsNull_*` — when `GetByOrgNo` returns `null`, `MapFromExternalIdenity` returns `null` and the service immediately calls `.IsPartyUuid(...)` on it — that's a `NullReferenceException` today. Decide whether to (a) write a failing test that pins the bug for a follow-up fix, or (b) skip until the service is hardened.

### Controller — `ConsentController.GetConsentStatusChanges`

The integration tests in `AccessMgmt.Tests/Controllers/Enterprise/ConsentControllerTestEnterpriseFetchStatusChanges.cs` cover auth (401/403), happy path, paging, and tie-breaking via the (legacy) `LegacyApiFixture`. New controller tests should prefer either:

- [ ] **Direct unit test** — instantiate `ConsentController` with `Mock<IConsent>` + a stubbed `ClaimsPrincipal`. Fast and enough for the controller's new branches: continuation-link building, `Unauthorized` when there's no party in the token, `result.Problem.ToActionResult()` propagation.
- [ ] **`ApiFixture`-based test** — only when the test needs the full MVC pipeline (model binding, auth policies, routing). See `docs/testing/FIXTURES.md`.

Do **not** add new consumers of `LegacyApiFixture` (see `docs/testing/FIXTURES.md` §3).

### Repository — `ConsentRepository.GetConsentStatusChangesForParty`

The implementation is mostly a SQL query plus base64 cursor parsing. Worth pinning down:

- [ ] Cursor parsing branch: invalid token → starts from beginning; valid token → resumes at the right offset.
- [ ] "Latest event per consentrequest" projection: multiple events per request, only the newest non-`created` one is returned.
- [ ] `ChangedDate` desc + `ConsentEventId` desc tie-breaker.
- [ ] Page-size clamping (out-of-range values fall back to the default of 100).

These need a real Postgres → write them in `Altinn.AccessMgmt.PersistenceEF.Tests` against `EFPostgresFactory` (template-cloned DB; ~100–500 ms per test). See `docs/testing/FIXTURES.md` "EFPostgresFactory" for the seeding strategy.

## Out of scope

The repository SQL query is already covered end-to-end by the existing integration test (auth, happy path, paging, tie-breaking via `LegacyApiFixture` in `ConsentControllerTestEnterpriseFetchStatusChanges.cs`); the repository-layer items above add focused projection/cursor/clamping coverage on top of that.

---
Parent #3486
Closes #3542

### References

- PR with the seed tests: [#2959](https://github.com/Altinn/altinn-authorization-tmp/pull/2959)
- Testing infra docs: [`docs/testing/`](https://github.com/Altinn/altinn-authorization-tmp/tree/main/docs/testing) — start with `WRITING_TESTS.md`, `FIXTURES.md`, `TEST_NAMING_CONVENTION.md`, `FLUENT_ASSERTIONS_GUIDELINES.md`.

### LINKED ISSUE #2976: Close pure-logic coverage gaps in Access Management API and persistence cores
## Goal

Close the pure-logic coverage gaps in Phase B: collect and complete the Tasks that bring under-covered non-DB assemblies up to expected line coverage. These are tests that can be written without a live database, Azurite, or other infrastructure, direct controller / service unit tests with mocked dependencies. Closing them frees up Phases C–F to focus on live-DB and threshold work.

## Background

The April 2026 audit of this repo's testing infrastructure ([`docs/testing/TESTING_INFRASTRUCTURE_OVERHAUL/TESTING_INFRASTRUCTURE_OVERHAUL_PART_2.md`](https://github.com/Altinn/altinn-authorization-tmp/blob/main/docs/testing/TESTING_INFRASTRUCTURE_OVERHAUL/TESTING_INFRASTRUCTURE_OVERHAUL_PART_2.md)) flagged several pure-logic assemblies whose line coverage is well below the levels we expect for non-DB code. Phase A (Feature #2946) closed the critical findings; this Feature collects the Tasks that close the pure-logic gaps in Phase B.

Audit findings covered. Numbers are post-Step-6 merged-cobertura baselines (see [`STEPS_PART_2/INDEX.md` § Final Coverage](https://github.com/Altinn/altinn-authorization-tmp/blob/main/docs/testing/TESTING_INFRASTRUCTURE_OVERHAUL/STEPS_PART_2/INDEX.md#final-coverage-measured)):

- **M6'** — controller gap: `Altinn.AccessManagement.Api.Metadata` at 51.53 % line / 51.11 % branch.
- **M5'** — domain pure-logic: `Altinn.AccessMgmt.Persistence.Core` at 25.39 % (largest pure-logic gap), `Altinn.AccessMgmt.Core` at 44.96 %.
- **M7'** — `Altinn.Authorization.Integration.Platform` at 54.94 % (`PaginatorStream<T>` + `TokenGenerator` where feasible without key-vault).

## Child tasks

Tasks are added here just-in-time, when the corresponding step is starting, not all upfront.

### Remaining

- [ ] **B.1 — Metadata controllers (M6')**: direct Moq tests for the remaining untested actions in `Altinn.AccessManagement.Api.Metadata`.
- [ ] **B.2 — Domain pure-logic (M5')**: pure-logic tests in `Altinn.AccessMgmt.Persistence.Core` and `Altinn.AccessMgmt.Core` (continues Part 1 Steps 42–60).
- [ ] **B.3 — Integration.Platform tail (M7')**: pure-logic tests for `PaginatorStream<T>` and (if reachable without key-vault) `TokenGenerator`.

---

Links: Audit doc [`TESTING_INFRASTRUCTURE_OVERHAUL_PART_2.md` § 4 Phase B](https://github.com/Altinn/altinn-authorization-tmp/blob/main/docs/testing/TESTING_INFRASTRUCTURE_OVERHAUL/TESTING_INFRASTRUCTURE_OVERHAUL_PART_2.md#4-phase-plan) · Step log [`STEPS_PART_2/INDEX.md` § Recommended Next Steps](https://github.com/Altinn/altinn-authorization-tmp/blob/main/docs/testing/TESTING_INFRASTRUCTURE_OVERHAUL/STEPS_PART_2/INDEX.md#recommended-next-steps-priority-order) items 6–8 · Predecessor Feature #2946 (Phase A — critical fixes; closed).

### LINKED ISSUE #3495: Investigate and re-enable skipped AccessManagement integration tests (hidden 500s in delegation)
## Problem

A test-hygiene audit (2026-06-14) of every `[Fact(Skip=...)]` / `[Theory(Skip=...)]` in the AccessManagement test suite. Several are **hidden failing integration tests** — they encode real 500s during delegation that are currently invisible because the test is skipped.

## Goal

These should be investigated and re-enabled (or the underlying defect fixed) before the production release.

### Hidden failures — need investigation (7)

`500 during delegation`:
- `Altinn.AccessManagement.Enduser.Api.Tests/Integration/Controllers/ConnectionsControllerTest.AddInstanceRights.cs:131`
- `.../ConnectionsControllerTest.RemoveInstance.cs:129`, `:178`
- `.../ConnectionsControllerTest.UpdateInstanceRights.cs:150`, `:317`
  (all: "Failing with 500 error during delegation - requires investigation")

`PDP returns 500 if party is missing`:
- `.../ClientDelegationControllerTest.cs:314`, `:549`

These return 500 where they should return a 4xx/handled result — likely a real product defect in the delegation path, not just a test issue. Root-cause and fix, then re-enable.

### Other skips

- `.../RequestControllerTest.cs:520` — "possible environmental issue", needs investigation.
- `.../RequestControllerTest.cs:736` — fixture mis-seeded (authenticates as a party with no delegation rights); needs a rewrite.
- `Altinn.Authorization.Host.Lease.Tests/StorageAccount/StorageAccountLeaseTest.cs:50,59` — legitimately require Azurite / a real storage account (the Lease.Tests project is excluded from the CI matrix). Keep skipped or wire Azurite into CI.

## Approach

Investigate each hidden-failure cluster, fix the underlying delegation 500s, and re-enable the tests. The Azurite-dependent ones are tracked separately (wire Azurite into CI or document the exclusion).

---
Parent #3486
Closes #3542

### LINKED ISSUE #3498: Add C# tests for Bruno-asserted authorization boundaries (PDP decisions + delegation guards)
## Problem

The Bruno API collections assert authorization **outcomes** (Permit / Deny / NotApplicable, and delegation-guard denials) that the C# automated suite barely covers — including privilege-escalation boundaries.

Surfaced by the 2026-06-14 discovery sweep. Prioritized by blast radius.

## Goal

Add C# tests mirroring these contracts so they're guarded in CI, not only in the manual Bruno suite.

## Work

- [ ] **1. `SystemResource` role→decision matrix (incl. NotApplicable)**
  Bruno: `src/apps/Altinn.Authorization/test/Bruno/.../Decision/SystemResource_Tests/*` (~18 requests). Asserts Permit vs **NotApplicable** for system resources (client-administration, enduser-access-management, main-administration, instance-delegation) per role (Dagl/Hadm/Kladm/Admai). The NotApplicable cells are negative privilege boundaries (a wrong Permit = escalation of the admin surface).
  - **Vehicle to decide first:** the C# PDP decision tests (`Altinn.Authorization.Tests/Integration/{AltinnApps,ResourceRegistry}_DecisionTests.cs`) are static policy+request+response fixtures; SystemResource decisions likely require role-graph resolution those fixtures don't exercise. Confirm whether to (a) author static fixtures or (b) use seeded AccessMgmt integration tests. Investigate before authoring 18 fixtures.
  - **DECISION (2026-06-17): descoped from C#.** These outcomes are decided by the system-resource XACML policies in the resource registry (external data: `altinn_client_administration`, `altinn_access_management`, `altinn_instance_delegation`, `altinn_access_management_hovedadmin`), not by code in this repo. The repo's XACML engine is already covered by `Altinn.Authorization.ABAC.Tests` + 36/36 conformance. A C# static-fixture port would require committing copies of the production policies (drift risk) and would mostly duplicate the Bruno suite, which already asserts this matrix against the live environment. Leaving area 1 to Bruno; C# effort goes to areas 2 and 5, which are code paths in this repo.

- [x] **2. `AM.VLD-00026` "package not assignable to recipient" (delegation-escalation guard)** — DONE (commit 487bfb8d)
  Bruno: `.../EnduserAPI/Connections/ToOthers/Negative_scenarioer/POST_Enduser_Conn_AccPck_AdmaiDelegatingPackageWithoutAccess.bru` (+ resource analog). Asserts 400 with code `AM.VLD-00026`, `User not authorized for operation.`, `paths: ["QUERY/packageId"]`.
  - **Guard:** `PackageValidation.AuthorizePackageAssignment` (AccessMgmt.Core/Validation/PackageValidation.cs:26) fails when the upstream delegation-check yields `Result=false`.
  - **Test home:** `ConnectionsControllerTest.AddAssignmentPackage.cs` (Enduser.Api.Tests). Existing tests use Jinx (MD, full delegable access) → all pass the guard. **Needs a limited actor (Admai or a rightholder) lacking delegable access to the target package**, so the delegation-check returns false. Requires seed-data setup; verify the denial is for the right reason.

- [ ] **3. Access-package decision matrix for persons**
  Bruno: `.../Decision/AccessPackageResource/*` + `.../Authorize/AccessPackageResource/*` (~25). Permit for a person reaching a resource via an access package — via key role, for a main unit, as rightholder, across every reportee identifier form. C# has only two **SystemUser** fixtures. Core production authz path. Integration tests (PDP, end-to-end resolution).

- [ ] **4. SystemUser client-delegation incl. `NoDelg → NotApplicable`**
  Bruno: `.../Decision/SystemUser/SysUser_ClientDelg_AccPkg_*` incl. the `NoDelg_NotApplicable` negative. Maskinporten, high blast radius. Integration tests.

- [x] **5. AuthorizedParties main-unit→sub-unit inheritance** — DONE (commit 980636bd)
  Bruno: `.../AuthorizedParties/.../GET_AuthorizedParties_Person_MainUnitDelgRolePcgRes_WithSubUnitPartyFilter.bru`. Contract: subunits inherit main-unit packages/resources/roles; `authorizedInstances` are **NOT** inherited; no duplicate parties; partyFilter on a subunit returns main+sub. The hovedenhet/underenhet regression class. C# (`AuthorizedPartiesControllerTest`) covers single-party presence but not these invariants. Integration test, seeded main+sub fixture.

## Out of scope

Area 1 is descoped from C# (see the DECISION note under area 1): external policy data, left to Bruno. Areas 3 & 4 are PDP decision-matrix tests (AccessPackageResource / SystemUser) whose Permit/NotApplicable outcomes are likewise decided by external resource-registry policies + the role graph, not code in this repo. Same drift-risk/duplication trade-off as area 1; candidates for the same descope, to be confirmed before any authoring.

## Constraints / principles

Each is authorization-correctness work; construct the scenarios carefully (negative cases must deny for the right reason). Do as a focused effort, not bundled with unrelated changes. Areas 1–2 are the escalation-boundary guards (highest priority); 3–5 widen coverage of the new access-package path.

## Findings

(2026-06-17, branch feat/3486_phase2_harden_cover)
- **Area 2 DONE** (commit 487bfb8d): negative `AddAssignmentPackage` test asserting 400 / AM.VLD-00026 (UserNotAuthorized) on `QUERY/packageId` for an actor without delegable access. The to-others package path was the gap (the client-delegation path already asserted UserNotAuthorized).
- **Area 5 DONE** (commit 980636bd): two `AuthorizedPartiesControllerTest` tests over the seeded Karlstad main/sub units: subunit nested under the main unit (no duplicate parties, carries the main unit's roles+packages) and partyFilter on the subunit returns main+sub. Pure main-only inheritance + instance-exclusion need a dedicated seed (the seeded actor holds a direct role on both units) and remain follow-ups.
- **Area 1 descoped** (see above): external policy data, left to Bruno.
- **Areas 3 & 4** are PDP decision-matrix tests (AccessPackageResource / SystemUser) whose Permit/NotApplicable outcomes are likewise decided by external resource-registry policies + the role graph, not code in this repo. Same drift-risk/duplication trade-off as area 1; candidates for the same descope, to be confirmed before any authoring.

---
Parent #3486
Closes #3542

### LINKED ISSUE #3543: ContextHandler.EnrichResourceParty throws on parties without a PartyUuid
## Problem

`ContextHandler.EnrichResourceParty` resolves a resource party from an organization number, a person id, or a party UUID, then copies the looked-up party's UUID into the resource attributes. Three of those assignments read `party.PartyUuid.Value` directly:

- the org-number branch (`PartyLookup(orgNo, null)`)
- the person-id branch (`PartyLookup(null, personId)`)
- the party-UUID branch (`GetPartiesAsync(List<Guid>)`)

`Party.PartyUuid` is nullable. When register returns a party without a UUID, `.Value` throws `InvalidOperationException` and enrichment fails for the whole authorization request.

The same method already guards the nullable everywhere else: the party-id fallback checks `PartyUuid.HasValue` before reading `.Value`, and the subject-resolution paths use `PartyUuid.HasValue ? .Value : Guid.Empty`. The three resource-party sites are the only ones that do not, so they are a latent null-dereference inconsistent with the rest of the file.

## Goal

Resource-party enrichment no longer throws when register returns a party without a UUID; the three resource-party sites match the safe pattern already used elsewhere in the method.

## Approach

Guard each of the three assignments with `HasValue`, matching the existing safe sites, and leave `PartyUuid` at `Guid.Empty` when register has no UUID.

## Work

- [ ] Guard the org-number branch (`PartyLookup(orgNo, null)`)
- [ ] Guard the person-id branch (`PartyLookup(null, personId)`)
- [ ] Guard the party-UUID branch (`GetPartiesAsync(List<Guid>)`)
- [ ] Pin with unit tests that drive each branch with a party that has a null UUID

Related to #3486. Fixed in PR #3542.

### LINKED ISSUE #3552: Report test coverage instead of gating it, and collect it only on the nightly scan
## Problem

CI test coverage was a PR-time gate: a drop below an assembly's target failed the build, and coverage was collected on every PR, adding instrumentation, merge and threshold-check cost to the critical path. The `dotnet-coverage` denominator also did not match SonarCloud's, so the two could disagree on the same code.

## Goal

Make coverage informational rather than a gate, and keep it off the PR critical path, while still measuring it against the same denominator as SonarCloud.

## Approach

- Align the `dotnet-coverage` denominator with SonarCloud (new `eng/testing/coverage.settings`) and mirror the exclusions into the duplication detector.
- Stop coverage from failing the build, locally and in CI. Below-target assemblies surface as warnings; the only hard failure left is a run that produces no coverage file at all.
- Collect coverage only on the nightly analyze run, not on every PR.

## Work

- [x] `eng/testing/coverage.settings`, `check-coverage-thresholds.ps1`, `run-coverage.ps1`, `coverage-thresholds.json`
- [x] `.github/workflows/tpl-vertical-ci.yml` collects coverage under the `analyze` input only
- [x] `docs/testing/COVERAGE.md`, `CI.md`, `SONARCLOUD.md` updated

## Out of scope

- SonarCloud server-side quality-gate configuration.
- The per-assembly target values themselves; this changes how they are enforced (reported, not gated), not what they are.

## Constraints / principles

Coverage percentage is a guardrail against regression, not a quality goal in itself, so a drop should be visible without blocking a merge.

Parent #3486. Shipped in PR #3542.

### LINKED ISSUE #3553: Break down integration-test DB-provision timing to locate setup-time cost
## Problem

`FixtureTiming` reported `db_provision` as a single opaque number, so the per-fixture provisioning cost could not be attributed: setup waste could not be told apart from real work.

## Goal

Decompose `db_provision` into named sub-phases so where integration-test setup time goes is visible per assembly, in CI.

## Approach

Add sub-phase buckets to the `===FIXTURE_TIMING===` line and time them at their sources:

- `server_start` (container acquire) and `provision_wait` (time blocked on the engine's provisioning lock) in `PostgresTestEngine`.
- `migrate` and `seed` (the two halves of the one-time template build) in `EFPostgresFactory.BuildTemplateAsync`.

## Work

- [x] New buckets emitted on the timing line
- [x] Surfaced automatically in the CI setup-timing breakdown
- [x] Finding recorded in `docs/testing/TEST_SETUP_TIMING.md`

## Out of scope

- Acting on the findings (shortening or overlapping the cold template build); that is the speed-up work in #3379.

## Findings

On the `AccessMgmt.Core.Tests` integration lane (4 fixtures): `db_provision` 68.0s = `provision_wait` 49.9s + `template_build` 12.5s (migrate 11.2 / seed 0.3) + `server_start` 4.1s + `clone` 1.5s. So `migrate` runs once (not per fixture) and clones are cheap (~0.4s): the template is already shared, and about 73% of the summed `db_provision` is contention behind the single cold template build at startup, not per-fixture work.

Parent #3379. Shipped in PR #3542.

### PR DESCRIPTION
**Pull request by howieandersen**
## Summary

Phase 2 "harden and cover" of the testing overhaul (#3486), in one PR. Large diff, small risk: the only product-behaviour change is two small null/parse fixes, each individually pinned. Everything else is behaviour-preserving test work, CI/coverage configuration, and docs. Test projects build clean and per-vertical CI on this branch is green.

## Results

| Area | Before | After |
|---|---|---|
| Register party with no UUID | `ContextHandler.EnrichResourceParty` threw `InvalidOperationException` and failed the whole authorization request (#3543) | nullable guarded at all three resource-party sites; pinned by per-branch tests |
| Malformed `systemuser_id` claim | `AuthenticationHelper.GetSystemUserUuid` threw `FormatException` | parses with `TryParse`, matching the sibling claim accessors; pinned |
| Delegation / request integration tests | 7 hidden 500s masked behind skips, plus other skips (#3495) | root-caused and re-enabled |
| Under-covered logic | `RightsHelper`, `StringExtensions`, `PackageValidation`, AccessMgmt.Core write-validation, the main-admin delegation-check fallback, the consent receiver-listing path, and PDP access-package decisions had no direct tests (#2976, #2975, #3498) | covered by real-logic and boundary tests, including new XACML decision fixtures |
| Coverage targets | primary tier of 7 assemblies; `Altinn.AccessManagement.Core` floor 60 | primary tier of 12; floors raised (`AccessManagement.Core` 60 to 75, `Authorization` 60 to 68, `Api.Enterprise` 60 to 67, `PersistenceEF` 90 to 94), five API assemblies added (Metadata, Enduser, Internal, ServiceOwner, Integration), and `AccessMgmt.Core` (53) + `Integration.Platform` (50) added to the warnings tier |
| Test-project build | build warnings across the verticals | 0 warnings |
| CI coverage denominator | diverged from SonarCloud | aligned via `eng/testing/coverage.settings` and `SonarQube.Analysis.xml`, with the exclusions mirrored into the duplication detector |
| Coverage regression | could fail the build | reported only, never gates (#3552) |
| Coverage collection | ran on every PR | nightly analyze run only (#3552) |
| `db_provision` setup timing | single opaque number | split into `server_start` / `migrate` / `seed` / `provision_wait`, surfaced in CI (#3553) |
| Testing docs + README | scattered, with stale gating wording and point-in-time numbers | brought to current state (`COVERAGE.md` slimmed, CI / SONARCLOUD / architecture / timing / fixtures / writing-tests docs updated); README SonarCloud badges linked to the per-project `branch=main` dashboards |

## Risk

The only product-behaviour changes are the two fixes in the first two rows. Both are small and individually pinned: each test was verified failing against the unfixed code and passing after. Everything else is tests, CI/coverage configuration, and docs, all behaviour-preserving. `Program.cs`/`Startup.cs` and `[CompilerGenerated]` code are deliberately kept in the coverage denominator (the settings file explains why).

## What's next

Phase 3 (consolidate; post-release, with soak time), tracked under #3486:
- #3379: speed up test runs (now scoped by the `provision_wait` finding above)
- #3412: consolidate test infrastructure across projects
- #3132: broader ABAC improvements

## Closes

**Production defect**
- Closes #3543

**Coverage**
- Closes #3498
- Closes #2976
- Closes #2975
- Closes #3495

**Test CI / infrastructure**
- Closes #3552
- Closes #3553

Related to #3486.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
...
+| **[Authorization](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization&branch=main)**     | [![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=Authorization_Authorization&metric=alert_status)](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization)       | [![Bugs](https://sonarcloud.io/api/project_badges/measure?project=Authorization_Authorization&metric=bugs)](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization)       | [![Code Smells](https://sonarcloud.io/api/project_badges/measure?project=Authorization_Authorization&metric=code_smells)](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization)       | [![Coverage](https://sonarcloud.io/api/project_badges/measure?project=Authorization_Authorization&metric=coverage)](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization)       | [![Duplicated Lines (%)](https://sonarcloud.io/api/project_badges/measure?project=Authorization_Authorization&metric=duplicated_lines_density)](https://sonarcloud.io/summary/new_code?id=Authorization_Authorization)       |
```
[the remark this answers is not in the document]
Comment (howieandersen) [reply README.md]: Done. Appended `&branch=main` to the badge click-through links too (commit 5abb9ed7), so every SonarCloud link in the table now lands on the explicit main view.

File: src/apps/Altinn.Authorization/test/Altinn.Authorization.Tests/Integration/ResourceRegistry_DecisionTests.cs [hunk 2]
```diff
@@ -363,6 +364,86 @@ public async Task PDP_Decision_ResourceRegistry_SystemUserWithDelegation_Returns
...
+        {
+            string testCase = "ResourceRegistry_SystemUserWithoutDelegation_NotApplicable";
+            HttpRequestMessage httpRequestMessage = TestSetupUtil.CreateJsonProfileXacmlRequest(testCase);
+            XacmlJsonResponse expected = TestSetupUtil.ReadExpectedJsonProfileResponse(testCase);
+
+            // Act
+            XacmlJsonResponse contextResponse = await TestSetupUtil.GetXacmlJsonProfileContextResponseAsync(_client, httpRequestMessage);
+
+            // Assert
+            AssertionUtil.AssertEqual(expected, contextResponse);
```
[the remark this answers is not in the document]
Comment (howieandersen) [reply src/apps/Altinn.Authorization/test/Altinn.Authorization.Tests/Integration/ResourceRegistry_DecisionTests.cs:383]: This path is not gated by the access-package feature flag, so `_client` is correct here. The `SystemUserWithDelegation_Permit` sibling uses the same `_client` and its expected response is Permit, which only occurs if the client-delegation decision path actually runs. The NotApplicable in this test therefore comes from the missing delegation, not from a disabled flag. The access-package flag gates the separate `SubjectWithoutAccessPackage` family, which does build its client via `GetTestClient(featureManager: ...)`.

**Review by jonkjetiloye — APPROVED** [review]

File: src/testing/FixtureTiming.cs [hunk 3]
```diff
@@ -65,6 +69,18 @@ internal enum Phase
 
         /// <summary>One-time container start + role bootstrap + migrate/seed of the template.</summary>
```
[the remark this answers is not in the document]
Comment (howieandersen) [reply src/testing/FixtureTiming.cs]: Fixed in 78bea8b2: the TemplateBuild doc no longer says container start (that is now the separate ServerStart phase).
